service/src/BatchServer.cpp

```diff
@@ -78,7 +79,20 @@ namespace geopm
             this->create_shmem();
         };
         auto run = [this]() {
-            this->run_batch();
+            try {
+                this->run_batch();
+            }
+            catch(...) {
+                if (m_is_client_waiting) {
```

Comment:
This looks redundant with the check inside `run_batch`, so one of these two loggers can probably be removed. This one _should_ be dead code in its current state since the other one clears `m_is_client_waiting` before rethrowing. 

I think removing either one would be fine. I'd probably favor removing the deeper-in-the-callstack log-and-rethrows since it should result in fewer instances of manually bubbling errors up a callstack.